### PR TITLE
Merge pull request #36 from aashen12/main

### PR DESCRIPTION
**Pull request by aashen12**
Merge pull request #35 from aashen12/development

### CODE REVIEW

File: R/mergeData.R [hunk 1]
```diff
@@ -32,20 +32,34 @@
...
 mergeData <- function(feat_eng_df, clinical_data, match = "ScrID", id_mapping = NULL) {
 
-  # note: still need to generalize this to more colnames.
   if (!is.null(id_mapping)) {
+
+  # converting literally everything into a factor
+  id_index <- grep(match, colnames(id_mapping))
+  id_mapping[, id_index] <- as.factor(id_mapping[, id_index])
+  feat_eng_df$pid <- as.factor(feat_eng_df$pid)
+  clinical_data$PatID <- as.factor(clinical_data$PatID)
```
Comment (aashen12) [inline R/mergeData.R]: How do you know that the patient ID column in `clinical_data` is `PatID`? I assume `pid` is the default output of the feature engineering df, correct?

File: R/kfold-cv.R [hunk 2]
```diff
@@ -7,7 +7,6 @@
 crossValidation <- function(clinical_data,
                             k = 5,
                             method = "coxph") {
```
Comment (aashen12) [inline R/kfold-cv.R]: The 3 possible options should be "coxph", "lasso", "ridge". Keep everything in lower case

File: R/kfold-cv.R [hunk 3]
```diff
@@ -7,7 +7,6 @@
...
-
   df <- clinical_data[-1]
   proportion <- 1-1/k
```
Comment (aashen12) [inline R/kfold-cv.R]: spaces

File: R/kfold-cv.R [hunk 4]
```diff
@@ -7,7 +7,6 @@
 crossValidation <- function(clinical_data,
                             k = 5,
                             method = "coxph") {
-
   df <- clinical_data[-1]
```
Comment (aashen12) [inline R/kfold-cv.R]: remove numeric dependence. Also, doing [-1] on a df is very confusing. You should try to use matrix subsetting

File: R/kfold-cv.R [hunk 5]
```diff
@@ -7,7 +7,6 @@
...
   proportion <- 1-1/k
```
Comment (aashen12) [inline R/kfold-cv.R]: In the `coxph` if statement, you should not grow the vector by initializing it as `c()`. You should initialize it as `rep(NA, k)`, where `k` is the number of folds. Then `cvector` is updated as `cvector[i] <- cindex`

File: R/kfold-cv.R [hunk 6]
```diff
@@ -27,7 +26,6 @@ crossValidation <- function(clinical_data,
     result <- list(score = mean(cvector),
                    concordances = cvector,
                    sd = sd(cvector))
-
   }
```
Comment (aashen12) [inline R/kfold-cv.R]: Also, the data should be partitioned such that each point gets to be in the test set once. It seems like you are just making k random partitions. We also need to do repeated CV with `n` repeats (see my comment above regarding this).

Chapter 5 of the `caret` manual gives you more detail of the `trainControl` and `train` functions which should achieve this: https://topepo.github.io/caret/model-training-and-tuning.html

The image below gives you an idea of what this looks like. In the image, there are 5 repeats with 6 folds (`number = 6`)
Screenshot 2023-11-25 at 9 56 39 AM

File: R/kfold-cv.R [hunk 7]
```diff
@@ -7,7 +7,6 @@
 crossValidation <- function(clinical_data,
```
Comment (aashen12) [inline R/kfold-cv.R]: We need to do repeated CV too. Please add another parameter `n` which controls the number of repeats (default 5).

File: R/kfold-cv.R [hunk 8]
```diff
@@ -27,7 +26,6 @@ crossValidation <- function(clinical_data,
     result <- list(score = mean(cvector),
                    concordances = cvector,
                    sd = sd(cvector))
-
   }
 
   else if (method == "LASSO") {
```
Comment (aashen12) [inline R/kfold-cv.R]: Again, see my comment above about repeated CV. The `cv.glmnet` portion is used for hyperparameter tuning. You should still use `train` and `trainControl` to do the outer CV loop. 

See image below. The inner loop is where we run `cv.glmnet`. I am currently looking up to see how this works with caret and will let you know when I find something. Otherwise we can chat on Monday.

Screenshot 2023-11-25 at 9 59 25 AM

Comment (aashen12) [reply R/kfold-cv.R]: The inner loop gives you the best `lambda`, and that `lambda` is used on the test set in the outer loop.

File: R/kfold-cv.R [hunk 9]
```diff
@@ -42,7 +40,9 @@ crossValidation <- function(clinical_data,
                        alpha = 1,
                        family = "cox",
                        type.measure = "C")
-    result <- cvfit
+    result <- list(score = mean(cvfit$cvm),
+                   concordances = cvfit$cvm,
+                   sd = sd(cvfit$cvm))
   }
   else if (method == "Ridge") {
```
Comment (aashen12) [inline R/kfold-cv.R]: All of my comments in the lasso section apply here

File: R/kfold-cv.R (hunk 6 above)
Comment (aashen12) [reply R/kfold-cv.R]: If for some reason we can't figure out a clean way to do this with the penalized models, then I have an alternative strategy for us to try.

File: R/kfold-cv.R (hunk 9 above)
Comment (aashen12) [inline R/kfold-cv.R]: Another reminder, we need to perform the feature selection within each CV fold. No need to worry about this quite yet but just something to keep in mind.

File: R/kfold-cv.R [hunk 10]
```diff
@@ -3,62 +3,136 @@
...
-      test <- df[-inTrain, ]
-      coxph_model <- coxph(Surv(time, status) ~., data = train)
-      test_pred <- coxph_model %>% predict(test)
-      cindex <- concordance.index(test_pred, test$time, test$status)$c.index
-      cvector <- c(cvector, cindex)
+    scores <- rep(NA, n)
+    for (j in 1:n) {
+      cvector <- rep(NA, k)
+      for (i in 1:k) {
+        inTrain <- createDataPartition(y = df$time, #figure out how to create actual partitions
```
Comment (aashen12) [inline R/kfold-cv.R]: Shouldn't the partitions be made outside of this loop? You are looping over each of the k partitions.
This comment also applies to ridge and lasso.
